firefox_puppeteer/api/software_update.py

```diff
@@ -0,0 +1,499 @@
+
+          let file = new FileUtils.File(arguments[0]);
+
+          let writer = iniFactory.createINIParser(file)
+                       .QueryInterface(Components.interfaces.nsIINIParserWriter);
+
+          writer.setString(arguments[1], arguments[2], arguments[3]);
+          writer.writeFile(null, Components.interfaces.nsIINIParserWriter.WRITE_UTF16);
+        """, script_args=[self.ini_file_path, self.INI_SECTION, self.INI_OPTION, new_channels])
+
+    def add_channels(self, channels):
+        """Add channels to the update-settings.ini file.
+
+        :param channels: an array of channel names to add
+        """
+        new_channels = set(self.channels) | set(channels)
+        self.channels = new_channels
+
+    def remove_channels(self, channels):
+        """Remove channels from the update-settings.ini file.
+
+        :param channels: an array of channel names to remove
+        """
+        self.channels = set(self.channels) - set(channels)
+
+
+class SoftwareUpdate(BaseLib):
+    """The SoftwareUpdate API adds support for an easy access to the update process."""
+    PREF_APP_DISTRIBUTION = 'distribution.id'
+    PREF_APP_DISTRIBUTION_VERSION = 'distribution.version'
+    PREF_APP_UPDATE_URL = 'app.update.url'
+    PREF_DISABLED_ADDONS = 'extensions.disabledAddons'
+
+    def __init__(self, marionette_getter):
+        BaseLib.__init__(self, marionette_getter)
+
+        self.app_info = AppInfo(marionette_getter)
+        self.prefs = Preferences(marionette_getter)
+
+        self._update_channel = UpdateChannel(marionette_getter)
+        self._mar_channels = MARChannels(marionette_getter)
+        self._active_update = ActiveUpdate(marionette_getter)
+
+    @property
+    def ABI(self):
+        """Get the customized ABI for the update service.
+
+        :returns: ABI version
+        """
+        abi = self.app_info.XPCOMABI
+        if mozinfo.isMac:
+            abi += self.marionette.execute_script("""
+              let macutils = Cc['@mozilla.org/xpcom/mac-utils;1']
+                             .getService(Components.interfaces.nsIMacUtils);
+              if (macutils.isUniversalBinary) {
+                return '-u-' + macutils.architecturesInBinary;
+              }
+              return '';
+            """)
+
+        return abi
+
+    @property
+    def active_update(self):
+        """ Holds a reference to an ActiveUpdate object."""
+        return self._active_update
+
+    @property
+    def allowed(self):
+        """Check if the user has permissions to run the software update
+
+        :returns: Status if the user has the permissions
+        """
+        allowed = self.marionette.execute_script("""
+          let aus = Cc['@mozilla.org/updates/update-service;1']
+                    .getService(Components.interfaces.nsIApplicationUpdateService);
+          return aus.canCheckForUpdates && aus.canApplyUpdates;
+        """)
+
+        return bool(allowed)
```

Comment:
`execute_script` already returns a boolean value, so you can directly return it.